### PR TITLE
Create new module for reusable unit test utilities

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose

$subject

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [X] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: features/admin.organizations.v1/__tests__/add-organization-modal.test.tsx [hunk 1]
```diff
@@ -40,8 +40,8 @@ const addComponentModalProps: AddOrganizationModalPropsInterface = {
 };
 
 describe("UTC-1.0 - [Organization Management Feature] - Add Organization Modal", () => {
-    const addOrganizationMock = jest.spyOn(api, "addOrganization");
-    const getOrganizationsMock = jest.spyOn(api, "getOrganizations");
+    const addOrganizationMock: any = jest.spyOn(api, "addOrganization");
```
Comment (brionmario) [inline features/admin.organizations.v1/__tests__/add-organization-modal.test.tsx:43]: See if `jest.SpyInstance` works here?

```suggestion
    const addOrganizationMock: : jest.SpyInstance = jest.spyOn(api, "addOrganization");
```

File: modules/unit-test-configs/package.json
Comment (brionmario) [inline modules/unit-test-configs/package.json:1]:
```json
{
  "private": true,
  "name": "@wso2is/unit-test-configs",
  "version": "0.1.0",
  "description": "Azure function app to handle routing requirements of Asgardeo Console Azure static app deployment.",
  "author": "WSO2",
  "license": "Apache-2.0",
  "type": "module",
  "main": "utils.tsx",
  "homepage": "https://github.com/wso2/identity-apps#README",
  "bugs": {
    "url": "https://github.com/wso2/identity-apps/issues"
  },
  "keywords": [
    "wso2",
    "identity-apps",
    "unit-testing"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "devDependencies": {
    "@asgardeo/auth-react": "^4.0.4",
    "@wso2is/admin.extensions.v1": "workspace:*",
    "@wso2is/admin.core.v1": "workspace:*",
    "msw": "^0.36.8",
    "react": "^18.2.0",
    "@wso2is/access-control": "workspace:*",
    "redux-mock-store": "^1.5.4",
    "redux": "^4.0.4"
  },
  "publishConfig": {
    "access": "restricted"
  }
}
```

**Review by brionmario — APPROVED** [review]

File: modules/unit-test-configs/package.json
Comment (pavinduLakshan) [reply modules/unit-test-configs/package.json:1]: Addressed with 3607cda

File: features/admin.organizations.v1/__tests__/add-organization-modal.test.tsx (hunk 1 above)
Comment (pavinduLakshan) [reply features/admin.organizations.v1/__tests__/add-organization-modal.test.tsx]: Addressed with 3607cda

**Comment by pavinduLakshan** [comment]
Uploading coverage reports to codecov fails, due to the GH action not being able to find the token.

**Review by brionmario — APPROVED** [review]

**Pull request merged**